presences/src/main/java/fr/openent/presences/controller/AlertController.java

```diff
@@ -88,24 +90,29 @@ public void getStudentsAlerts(HttpServerRequest request) {
         List<String> classes = request.params().getAll(Field.CLASS);
         String startAt = request.params().get(Field.START_AT);
         String endAt = request.params().get(Field.END_AT);
+        Integer page = request.params().get(Field.PAGE) != null ? Integer.parseInt(request.getParam(Field.PAGE)) : 0;
         if (types.size() == 0) {
             badRequest(request);
             return;
         }
-        getAlerts(request, types, students, classes, startAt, endAt, arrayResponseHandler(request));
+        getAlerts(request, types, students, classes, startAt, endAt, page, arrayResponseHandler(request));
     }
 
     private void getAlerts(HttpServerRequest request, List<String> types, List<String> students, List<String> classes,
-                           String startDate, String endDate, Handler<Either<String, JsonArray>> handler) {
+                           String startDate, String endDate, Integer page, Handler<Either<String, JsonArray>> handler) {
         Future<JsonArray> groupStudentFuture = (classes.isEmpty()) ? Future.succeededFuture(new JsonArray()) : groupService.getGroupStudents(classes);
-        groupStudentFuture.compose(users -> {
-                    users.stream()
-                            .map(o -> (JsonObject)o)
-                            .map(jsonObject -> jsonObject.getString(Field.ID))
-                            .forEach(students::add);
-                    return alertService.getAlertsStudents(request.getParam(Field.ID), types, students, startDate, endDate, "00:00:00", "23:59:59");
+        Future<Integer> pageCountFuture = alertService.getPageCount(request.getParam(Field.ID), types, students, startDate, endDate, "00:00:00", "23:59:59");
+        CompositeFuture.all(groupStudentFuture,pageCountFuture)
+                        .compose(futures -> {
+                            groupStudentFuture.result().stream()
+                                    .map(o -> (JsonObject)o)
```

Comment:
```suggestion
                            .filter(JsonObject.class::isInstance)
                            .map(JsonObject.class::cast)
```
(sinon y'en a qui seront pas content 😋)